server/src/main/java/org/elasticsearch/index/mapper/LuceneDocument.java

```diff
@@ -36,7 +36,7 @@
      * for generating the _tsid field. The map will be used by {@link TimeSeriesIdFieldMapper}
      * to build the _tsid field for the document.
      */
-    private SortedMap<String, BytesReference> dimensionBytes;
+    private SortedMap<BytesRef, BytesReference> dimensionBytes;
```

Comment:
I switched to bytes ref here to make it really obvious these are sorted by utf-8.

Reply:
Ulterior motive: I'll want them in utf-8 in more than one place.